File: 2211_03/바이러스.py
# https://www.acmicpc.net/problem/2606
import sys, collections
N = int(sys.stdin.readline().strip())
M = int(sys.stdin.readline().strip())
PAIRS = collections.defaultdict(list)
for _ in range(M):
    n1, n2 = map(int, sys.stdin.readline().split())
    PAIRS[n1].append(n2)
    PAIRS[n2].append(n1)

cnt = 0
q = collections.deque()
q.append(1)
visited = [False] * (N + 1)
visited[1] = True
while q:
    n = q.popleft()
    for i in PAIRS[n]:
        if not visited[i]:
            cnt += 1
            visited[i] = True
            q.append(i)
print(cnt)

File: 2211_03/바이러스.java
// https://www.acmicpc.net/problem/2606
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class Main {
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int M = sc.nextInt();
        int[][] PAIRS = new int[101][101];
        for(int i=0;i<M;i++){
            int n1 = sc.nextInt();
            int n2 = sc.nextInt();
            PAIRS[n1][n2] = PAIRS[n2][n1] = 1;
        }

        int cnt = 0;
        int[] visited = new int[101];
        Queue<Integer> q = new LinkedList<>();
        q.add(1);
        visited[1] = 1;
        while(!q.isEmpty()){
            int n = q.poll();
            for(int i=1;i<N+1;i++){
                if(PAIRS[n][i] == 1 && visited[i] == 0){
                    q.add(i);
                    visited[i] = 1;
                    cnt += 1;
                }
            }
        }
        System.out.println(cnt);
    }
}
